google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -866,6 +975,71 @@ private void cleanupInflightRequests() {
     }
   }
 
+  private Boolean retryOnRetryableError(Code errorCode, AppendRequestAndResponse requestWrapper) {
+    if (this.maxRetryNumAttempts == 0) {
+      return false;
+    }
+
+    if (!isImmediatelyRetriableStatus(errorCode) && errorCode != Code.RESOURCE_EXHAUSTED) {
```

Comment:
why this is split out?

Reply:
RESOURCE_EXHAUSTED errors are not immediately retryable, so we'd have to check that as well.